### PR TITLE
mkosi: add page

### LINKED ISSUE #10692: page request: mkosi
### Command description

Tool for building os images described as

> A fancy wrapper around `dnf --installroot`, `apt`, `pacman` and `zypper` that generates customized disk images with a number of bells and whistles.

Part of systemd

https://github.com/systemd/mkosi

### Command details

v16

### OS

- [ ] Android
- [X] Linux
- [ ] macOS
- [ ] sunOS
- [ ] Windows

### PR DESCRIPTION
**Pull request by fwilhe2**
- [x] The page(s) are in the correct platform directories: `common`, `linux`, `osx`, `windows`, `sunos`, `android`, etc.
- [x] The page(s) have at most 8 examples.
- [x] The page description(s) have links to documentation or a homepage.
- [x] The page(s) follow the [content guidelines](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#guidelines).
- [x] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#commit-message).
- **Version of the command being documented (if known):** v16

Closes #10692

### CODE REVIEW

**Comment by fwilhe2** [comment]
This is a draft because I'm not sure what the correct use of 'sudo' in this case is. Which commands require sudo may depend on the system config as far as I'm aware. I don't want to preface all commands with sudo, but I don't know how to properly mark that.

Any suggestion to improve this is welcome.

**Comment by blueskyson** [comment]
> This is a draft because I'm not sure what the correct use of 'sudo' in this case is. Which commands require sudo may depend on the system config as far as I'm aware. I don't want to preface all commands with sudo, but I don't know how to properly mark that.
> 
> Any suggestion to improve this is welcome.

You can add sudo prefaces if the command always requires superuser permission. But it's not mandatory.

File: pages/linux/mkosi.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# mkosi
+
+> Tool for building modern, legacy-free Linux images. Part of systemd.
+> More information: <https://github.com/systemd/mkosi>.
```
Comment (blueskyson) [inline pages/linux/mkosi.md:4]: manned.org's page is more useful.
```suggestion
> More information: <https://manned.org/mkosi>.
```

**Comment by fwilhe2** [comment]
> > This is a draft because I'm not sure what the correct use of 'sudo' in this case is. Which commands require sudo may depend on the system config as far as I'm aware. I don't want to preface all commands with sudo, but I don't know how to properly mark that.
> > Any suggestion to improve this is welcome.
> 
> You can add sudo prefaces if the command always requires superuser permission. But it's not mandatory.

okay, I'd just leave it as is

**Comment by fwilhe2** [comment]
Should be ready to merge from my perspective

File: pages/linux/mkosi.md
Comment (kant) [inline pages/linux/mkosi.md:1]: line 10: `distribuion`

Comment (fwilhe2) [reply pages/linux/mkosi.md:1]: good catch, thanks.

Comment (kant) [reply pages/linux/mkosi.md:1]: ok!

File: pages/linux/mkosi.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# mkosi
+
+> Tool for building modern, legacy-free Linux images. Part of systemd.
```
Comment (kbdharun) [inline pages/linux/mkosi.md:3]:
```suggestion
> Tool for building modern, legacy-free Linux images.
> Part of `systemd`.
```

File: pages/linux/mkosi.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+- Build an image with default settings, if no distribution is selected, distribution of the host system is used:
+
+`mkosi build --distribution={{fedora|debian|ubuntu|arch|opensuse}}`
+
+- Build image and get interactive shell in a systemd-nspawn container of the image:
+
+`mkosi shell`
+
+- Boot image in a virtual machine using qemu, only supported for disk images or cpio images when a kernel is provided:
```
Comment (kbdharun) [inline pages/linux/mkosi.md:18]:
```suggestion
- Boot image in a virtual machine using QEMU, only supported for disk images or CPIO images when a kernel is provided:
```

File: pages/linux/mkosi.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+`mkosi qemu`
+
+- Get more help:
```
Comment (kbdharun) [inline pages/linux/mkosi.md:22]:
```suggestion
- Display help:
```
We use this generic text for help descriptions across our pages.

**Review by kbdharun — COMMENTED** [review]
Thanks for your contribution. I have some suggestions for this page after this the PR is GTG.

**Review by kbdharun — APPROVED** [review]

**Comment by fwilhe2** [comment]
> Thanks for your contribution. I have some suggestions for this page after this the PR is GTG.

thanks :)

File: pages/linux/mkosi.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+> Tool for building modern, legacy-free Linux images.
+> Part of `systemd`.
+> More information: <https://manned.org/mkosi>.
+
+- Show current build configuration to verify what would be built:
+
+`mkosi summary`
+
+- Build an image with default settings, if no distribution is selected, distribution of the host system is used:
```
Comment (acuteenvy) [inline pages/linux/mkosi.md:11]:
```suggestion
- Build an image with default settings (if no distribution is selected, the distribution of the host system is used):
```

File: pages/linux/mkosi.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+`mkosi build --distribution={{fedora|debian|ubuntu|arch|opensuse}}`
```
Comment (acuteenvy) [inline pages/linux/mkosi.md:13]:
```suggestion
`mkosi build --distribution {{fedora|debian|ubuntu|arch|opensuse|...}}`
```

File: pages/linux/mkosi.md [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+- Build image and get interactive shell in a systemd-nspawn container of the image:
```
Comment (acuteenvy) [inline pages/linux/mkosi.md:15]:
```suggestion
- Build an image and run an interactive shell in a systemd-nspawn container of the image:
```

File: pages/linux/mkosi.md [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+
+`mkosi shell`
+
+- Boot image in a virtual machine using QEMU, only supported for disk images or CPIO images when a kernel is provided:
```
Comment (acuteenvy) [inline pages/linux/mkosi.md:19]:
```suggestion
- Boot an image in a virtual machine using QEMU (only supported for disk images or CPIO images when a kernel is provided):
```

**Review by kbdharun — APPROVED** [review]

**Review by acuteenvy — APPROVED** [review]

**Pull request merged**